pac4j-core/src/main/java/org/pac4j/core/profile/UserProfile.java

```diff
@@ -257,10 +257,10 @@ public boolean containsAuthenicationAttribute(final String name) {
      * @param <T> The type of the authentication attribute
      * @return the named attribute
      */
-    public <T> T getAuthenticationAttribute(final String name, final Class<T> clazz)
+    public <T> Optional<T> getAuthenticationAttribute(final String name, final Class<T> clazz)
```

Comment:
Why not add an `Optional` to the `getAttribute` and `getAttributeByType` methods?